### PR TITLE
Add USGS LCMAP

### PR DESCRIPTION
**Pull request by pjhartzell**
### USGS Land Change, Monitoring, Assessment, and Projection (LCMAP)

**Home:** https://www.usgs.gov/special-topics/lcmap

LCMAP provides land cover and land cover change monitoring products derived from Landsat Analysis Ready Data (ARD). This PR provides the latest LCMAP Collections for CONUS (Collection 1.3, 1985-2021) and Hawaii (Collection 1.0, 2000-2020). Note that a time-series model underpins the USGS's change detection and thematic classification algorithm. The effect is that classification values for many of the years can change each time a new year is added to the dataset. Therefore, a new USGS LCMAP Collection is generated each time a new year is added to the dataset.

The USGS provides LCMAP data in tiles according to the Landsat ARD tile grid. The data is provided as TAR archives, which contain COGs and XML metadata for 10 Science Products. The COGs are reprocessed via the [LCMAP stactools package](https://github.com/stactools-packages/usgs-lcmap) to add overviews, "nodata" values where appropriate, and an updated projection definition. The source USGS data bundle is also included as an Asset on the STAC Items.

### STAC Collections:
- [usgs-lcmap-conus-v13](https://pct-apis-staging.westeurope.cloudapp.azure.com/stac/collections/usgs-lcmap-conus-v13)
- [usgs-lcmap-hawaii-v10](https://pct-apis-staging.westeurope.cloudapp.azure.com/stac/collections/usgs-lcmap-hawaii-v10)

### PC Test
- [Dataset group page](https://planetarycomputer-test.microsoft.com/dataset/group/usgs-lcmap) 
- [Data Explorer View](https://planetarycomputer-test.microsoft.com/explore?c=-122.3712%2C40.6806&z=8.00&v=2&d=usgs-lcmap-conus-v13&m=Most+recent+%282021%29&r=Classification&s=false%3A%3A100%3A%3Atrue&sr=desc&ae=0)
image

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

All of the Hawaii Collection and 1000 Items in the CONUS Collection have been ingested into PC Test.

## Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review
- [ ] Changelog has been updated
- [ ] Documentation has been updated
- [ ] Unit tests pass locally (./scripts/test)
- [x] Code is linted and styled (./scripts/format)

### CODE REVIEW

File: datasets/usgs-lcmap/collection/usgs-lcmap-conus-v13/description.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
...
+
+This [STAC](https://stacspec.org/en) Collection contains [LCMAP CONUS Collection 1.3](https://www.usgs.gov/special-topics/lcmap/collection-13-conus-science-products), which was released in August 2022 for years 1985-2021. The data are tiled according to the Landsat ARD tile grid and consist of [Cloud Optimized GeoTIFFs](https://www.cogeo.org/) (COGs) and corresponding metadata files. Note that the provided COGs differ slightly from those in the USGS source data. They have been reprocessed to add overviews, "nodata" values where appropriate, and an updated projection definition. The source USGS data bundle is also provided in TAR format.
```
Comment (TomAugspurger) [inline datasets/usgs-lcmap/collection/usgs-lcmap-conus-v13/description.md:3]:
```suggestion
This [STAC](https://stacspec.org/en) Collection contains [LCMAP CONUS Collection 1.3](https://www.usgs.gov/special-topics/lcmap/collection-13-conus-science-products), which was released in August 2022 for years 1985-2021. The data are tiled according to the Landsat ARD tile grid and consist of [Cloud Optimized GeoTIFFs](https://www.cogeo.org/) (COGs) and corresponding metadata files. Note that the provided COGs differ slightly from those in the USGS source data. They have been reprocessed to add overviews, "nodata" values where appropriate, and an updated projection definition.
```

File: datasets/usgs-lcmap/collection/usgs-lcmap-hawaii-v10/description.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+
+This [STAC](https://stacspec.org/en) Collection contains [LCMAP Hawaii Collection 1.0](https://www.usgs.gov/special-topics/lcmap/collection-1-hawaii-science-products), which was released in January 2022 for years 2000-2020. The data are tiled according to the Landsat ARD tile grid and consist of [Cloud Optimized GeoTIFFs](https://www.cogeo.org/) (COGs) and corresponding metadata files. Note that the provided COGs differ slightly from those in the USGS source data. They have been reprocessed to add overviews, "nodata" values where appropriate, and an updated projection definition. The source USGS data bundle is also provided in TAR format.
```
Comment (TomAugspurger) [inline datasets/usgs-lcmap/collection/usgs-lcmap-hawaii-v10/description.md:3]:
```suggestion
This [STAC](https://stacspec.org/en) Collection contains [LCMAP Hawaii Collection 1.0](https://www.usgs.gov/special-topics/lcmap/collection-1-hawaii-science-products), which was released in January 2022 for years 2000-2020. The data are tiled according to the Landsat ARD tile grid and consist of [Cloud Optimized GeoTIFFs](https://www.cogeo.org/) (COGs) and corresponding metadata files. Note that the provided COGs differ slightly from those in the USGS source data. They have been reprocessed to add overviews, "nodata" values where appropriate, and an updated projection definition.
```

File: datasets/usgs-lcmap/collection/usgs-lcmap-hawaii-v10/template.json [hunk 3]
```diff
@@ -0,0 +1,1527 @@
...
+            "interval": [
+                [
+                    "2000-01-01T00:00:00Z",
+                    "2020-12-31T00:00:00Z"
+                ]
+            ]
+        }
+    },
+    "item_assets": {
+        "tar": {
```
Comment (TomAugspurger) [inline datasets/usgs-lcmap/collection/usgs-lcmap-hawaii-v10/template.json:109]: Will remove this and `tar_metadata`

File: datasets/usgs-lcmap/collection/usgs-lcmap-conus-v13/template.json [hunk 4]
```diff
@@ -0,0 +1,1541 @@
...
+            "interval": [
+                [
+                    "1985-01-01T00:00:00Z",
+                    "2021-12-31T00:00:00Z"
+                ]
+            ]
+        }
+    },
+    "item_assets": {
+        "tar": {
```
Comment (TomAugspurger) [inline datasets/usgs-lcmap/collection/usgs-lcmap-conus-v13/template.json:117]: Will remove this and "tar_metadata"

**Review by TomAugspurger — APPROVED** [review]
As discussed elsewhere, we won't be hosting the TAR of COGs fetched from the USGS API. Left a couple notes of places that'll need to be updated (along with `create_item`).

LGTM other than that.

**Comment by pjhartzell** [comment]
@TomAugspurger The removal of all mentions of the `tar` and `tar_metadata` assets is complete.

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**